packages/volto-slate/src/editor/less/slate.less

```diff
@@ -5,19 +5,27 @@ h4 {
   &:hover {
     a.anchor {
       svg {
-        visibility: unset;
+        opacity: 1;
+        transform: rotate(15deg);
       }
     }
   }
 
+  .anchor-wrapper {
+    margin-left: 5px;
+    vertical-align: middle;
+  }
+
   a.anchor {
```

Comment:
It's possible to construct a heading with just the right (wrong?) length so that the anchor svg flows to the next line and takes up additional vertical space. To avoid that, I'd suggest adding `position: absolute;` so that the anchor never affects positioning! It means it might flow into the margin in this edge case, but that's probably not a big problem.

Reply:
I was having problems with the alignment of anchor with title. For example if text is "aligned"(center,left) or if font is too big...etc. Anyway, I pushed your suggestions in [c33f634](https://github.com/plone/volto/pull/4287/commits/c33f634e32bcd7eb56ae80d36dd5258f2aeb101e)

Reply:
It's certainly possible there are still edge cases, but it looks okay to me now.